src/main/java/com/mycompany/app/App.java

```diff
@@ -17,6 +17,12 @@ public static void main( String[] args )
     private void foo() throws Exception {
         bar("DES"); // this should create a warning
     }
+    
+    private void foo2() throws Exception {
+        if ("A" == "B") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.